hacspec/src/lib.rs

```diff
@@ -855,46 +877,61 @@ fn parse_message_1(
     EDHOCError,
 > {
     let mut error: EDHOCError = EDHOCError::UnknownError;
+    let mut method: U8 = U8(0xff);
     let mut g_x: BytesP256ElemLen = BytesP256ElemLen::new();
     let mut suites_i = BytesSuites::new();
     let mut suites_i_len: usize = 0;
     let mut raw_suites_len: usize = 0;
     let mut c_i = U8(0);
     let mut ead_1 = None::<EADItemHacspec>;
 
-    let method = rcvd_message_1.content[0];
+    // first element of CBOR sequence must be an integer
+    if is_cbor_uint8(rcvd_message_1.content[0]) {
+        method = rcvd_message_1.content[0];
 
-    let res_suites = parse_suites_i(rcvd_message_1);
+        let res_suites = parse_suites_i(rcvd_message_1);
 
-    if res_suites.is_ok() {
-        (suites_i, suites_i_len, raw_suites_len) = res_suites.unwrap();
+        if res_suites.is_ok() {
+            (suites_i, suites_i_len, raw_suites_len) = res_suites.unwrap();
 
-        g_x = BytesP256ElemLen::from_slice(
-            &rcvd_message_1.content,
-            3 + raw_suites_len,
-            P256_ELEM_LEN,
-        );
-
-        c_i = rcvd_message_1.content[3 + raw_suites_len + P256_ELEM_LEN];
+            let g_x_type = rcvd_message_1.content[1 + raw_suites_len].declassify();
```

Comment:
You could use the similar check here as for `uint8`

Reply:
Yes I will do it on all other parsing and decoding routines.